lib/seattleflu/id3c/cli/command/etl/redcap_det_uw_reopening.py

```diff
@@ -0,0 +1,1210 @@
+            event_type = EventType.ENROLLMENT
+        elif redcap_record_instance["redcap_event_name"] == ENCOUNTER_EVENT_NAME:
+            event_type = EventType.ENCOUNTER
+        else:
+            LOG.error(f"The record has an unexpected event name: {redcap_record_instance['redcap_event_name']}")
+            continue
+
+        if event_type == EventType.ENCOUNTER:
+            if is_complete('kiosk_registration_4c7f', redcap_record_instance):
+                collection_method = CollectionMethod.KIOSK
+            elif is_complete('test_fulfillment_form', redcap_record_instance):
+                collection_method = CollectionMethod.SWAB_AND_SEND
+
+        # site_reference refers to where the sample was collected
+        site_reference = create_site_reference(redcap_record_instance, collection_method, event_type)
+
+        initial_encounter_entry, initial_encounter_reference = create_encounter(
+            redcap_record_instance, patient_reference, site_reference,
+            location_resource_entries, event_type, collection_method)
+
+        # Skip the entire record if we can't create the enrollment encounter.
+        # Otherwise, just skip the record instance.
+        if not initial_encounter_entry:
+            if event_type == EventType.ENROLLMENT:
+                return None
+            else:
+                LOG.warning("Skipping record instance with insufficient information to construct a initial encounter")
+                continue
+
+        specimen_entry = None
+        specimen_observation_entry = None
+        specimen_received = (collection_method == CollectionMethod.SWAB_AND_SEND and \
+            is_complete('post_collection_data_entry_qc', redcap_record_instance)) or \
+            (collection_method == CollectionMethod.KIOSK and \
+            is_complete('kiosk_registration_4c7f', redcap_record_instance))
+
+        if specimen_received:
+            specimen_entry, specimen_reference = create_specimen(redcap_record_instance, patient_reference, collection_method)
+            specimen_observation_entry = create_specimen_observation_entry(
+                specimen_reference, patient_reference, initial_encounter_reference)
+        else:
+            LOG.info("Creating encounter for record instance without sample")
+
+        if specimen_received and not specimen_entry:
+            LOG.warning("Skipping record instance with insufficent information to construct a specimen")
+            continue
+
+        computed_questionnaire_entry = None
+        enrollment_questionnaire_entry = None
+        daily_questionnaire_entry = None
+        testing_determination_internal_questionnaire_entry = None
+
+        computed_questionnaire_entry = create_computed_questionnaire_response(
+            redcap_record_instance, patient_reference, initial_encounter_reference,
+            birthdate, datetime.strptime(initial_encounter_entry['resource']['period']['start'], '%Y-%m-%d'))
```

Comment:
`datetime.strptime(initial_encounter_entry['resource']['period']['start'], '%Y-%m-%d')` seems like something the `create_computed_questionnaire_response()` function can know how to do rather than performing a complicated parse on the fly when passing this value as an argument.

Reply:
The parameters needed for this bit of create_computed_questionnaire_response are:
birthdate: datetime
encounter_date: datetime
The function needs the two dates so it can compute the participant's age at the time of the encounter.

The logic that you pointed out is needed so the caller can pass in the encounter_date.

create_computed_questionnaire_response does take a encounter_reference: dict parameter , but I did not want to have it create the encounter date itself.